std/os/uefi/protocols/simple_text_input_ex_protocol.zig

```diff
@@ -0,0 +1,74 @@
+const uefi = @import("std").os.uefi;
+const Event = uefi.Event;
+const Guid = uefi.Guid;
+
+/// UEFI Specification, Version 2.8, 12.3
+pub const SimpleTextInputExProtocol = extern struct {
+    _reset: extern fn (*const SimpleTextInputExProtocol, u8) usize,
+    _read_key_stroke_ex: extern fn (*const SimpleTextInputExProtocol, *KeyData) usize,
+    wait_for_key_ex: *Event,
+    _set_state: extern fn (*const SimpleTextInputExProtocol, *const u8) usize,
+    _register_key_notify: extern fn (*const SimpleTextInputExProtocol, *const KeyData, extern fn (*const KeyData) usize, **c_void) usize,
+    _unregister_key_notify: extern fn (*const SimpleTextInputExProtocol, *const c_void) usize,
+
+    pub fn reset(self: *const SimpleTextInputExProtocol, verify: bool) usize {
+        return self._reset(self, @boolToInt(verify));
+    }
+
+    pub fn readKeyStrokeEx(self: *const SimpleTextInputExProtocol, key_data: *KeyData) usize {
+        return self._read_key_stroke_ex(self, key_data);
+    }
+
+    pub fn setState(self: *const SimpleTextInputExProtocol, state: *const u8) usize {
+        return self._set_state(self, state);
+    }
+
+    pub fn registerKeyNotify(self: *const SimpleTextInputExProtocol, key_data: *const KeyData, notify: extern fn (*const KeyData) usize, handle: **c_void) usize {
+        return self._register_key_notify(self, key_data, notify, handle);
+    }
+
+    pub fn unregisterKeyNotify(self: *const SimpleTextInputExProtocol, handle: *const c_void) usize {
+        return self._unregister_key_notify(self, handle);
+    }
+
+    pub const guid align(8) = Guid{ 0x34, 0x75, 0x9e, 0xdd, 0x62, 0x77, 0x98, 0x46, 0x8c, 0x14, 0xf5, 0x85, 0x17, 0xa6, 0x25, 0xaa };
+};
+
+pub const KeyData = extern struct {
+    key: InputKey,
+    key_state: KeyState,
+
+    pub fn init() KeyData {
+        return KeyData{
+            .key = undefined,
+            .key_state = undefined,
+        };
+    }
+};
+
+pub const KeyState = extern struct {
+    key_shift_state: u32,
```

Comment:
Should this be a struct of bits?

Reply:
Like this? I'm not sure if we would lose alignment guarantees.
```zig
pub const KeyState = extern struct {
    key_shift_state: packed struct {
        left_logo_pressed: bool,
        right_logo_pressed: bool,
        left_alt_pressed: bool,
        right_alt_pressed: bool,
        left_control_pressed: bool,
        right_control_pressed: bool,
        left_shift_pressed: bool,
        right_shift_pressed: bool,
        _pad1: u6,
        sys_req_pressed: bool,
        menu_key_pressed: bool,
        _pad2: u15,
        shift_state_valid: bool,
    },
    key_toggle_state: packed struct {
        toggle_state_valid: bool,
        _pad1: u2,
        key_state_exposed: bool,
        caps_lock_active: bool,
        _pad2: u1,
        num_lock_active: bool,
        scroll_lock_active: bool,
    },
};
```

Reply:
Yeah that's the sort of thing I meant.
Until #1512  is complete you have to specify alignment at the variable definition site (rather than in the type)

Reply:
I guess because this structure is only ever returned from UEFI and never passed to it by the user, we don't have to worry about it.

Reply:
New commit looks good :)